enhancement(log): show stashes in log view, toggle visibility in view options popover

calahatras opened the pull request.

calahatras committed:
enhancement(log): show stashes in log view, toggle visibility in view options popover

calahatras committed:
wip(log): parse stashes similar as log events, display list of stashes differently using subject text

mickedefault commented on GitOut/Features/Git/LocalGitRepository.cs:
```diff
@@ -113,57 +113,25 @@ public async Task<IEnumerable<GitHistoryEvent>> LogAsync(LogOptions options)
-                        break;
+                    if (historyByCommitId.TryGetValue(item.ParentId!, out GitHistoryEvent? parent))
+                    {
+                        item.Branches.Add(GitBranchName.CreateLocal($"stash@{{{i}}}"));
```
can we use the stash reflog selector as the branch name? [git pretty formats](https://git-scm.com/docs/pretty-formats)
>%gD
> reflog selector, e.g., refs/stash@{1} or refs/stash@{2 minutes ago}; [..]
> %gd
> shortened reflog selector; same as %gD, but the refname portion is shortened for human readability (so refs/heads/master becomes just master).

calahatras replied:
You mean just prefix with `refs/`?

mickedefault replied:
hm. I can't remember why I figured that name would be better. 🤔 
what i was after was the descriptive name (for instance "WIP on <branch>: <description>")

calahatras replied:
The "WIP on" text is available with the latest change. I update GitStash to be a subclass of GitLogEntry instead, and read stashes using the same pretty format as logs, which means we get author, dates, and all that info as well.

mickedefault commented on GitOut/Features/Git/LogOptions.cs:
```diff
@@ -2,11 +2,7 @@ namespace GitOut.Features.Git
 {
     public class LogOptions
```
can be converted to a record

calahatras replied:
It is now

mickedefault reviewed (changes requested):
~unfortunately crashes at the moment when opening a project without stashes.~

calahatras committed:
refactor(stash): add branch name to stash, only build a log item if any data has been read from log output

calahatras committed:
clean(stash): remove test since GitStash is now created using a GitLogEntryBuilder

calahatras committed:
refactor(log): add icon to stash branches, make LogOptions a record

calahatras committed:
clean(log): remove status from canvas tooltip

mickedefault committed:
enhancement(log): renders log paths based on line properties

cleanup is required before merging to main. this is here to test and validate

mickedefault committed:
enhancement(log): use triangles instead of rombs for stashes

mickedefault committed:
fix(log): tweaking ui for different line types/commit shapes

there is still an issue with the initial node looking
like a stash, since it's bottom line is missing.
a shape/variant/committype should most likely be
able to solve that, and checking that instead of
the linetype when painting the commit shape

calahatras committed:
cleanup(log): remove unused converter, remove empty line break

mickedefault committed:
refactor(log): NodesToPathConverter cleanup

calahatras committed:
cleanup(collections): remove unused using statement

mickedefault committed:
refactor(log): NodesToPathConverter cleanup

mickedefault committed:
fixup! refactor(log): NodesToPathConverter cleanup

mickedefault reviewed (approved):
:star:
